lib/helpers/hearing_renderer.rb

```diff
@@ -0,0 +1,335 @@
+    else
+      type_conversions = get_appeal_type_conversions(appeal)
+      type_conversions_with_original_type = add_original_type(
+        appeal.readable_original_hearing_request_type,
+        type_conversions
+      )
+      text = format_original_and_current_type(appeal)
+      text += format_conversions(type_conversions_with_original_type)
+      text
+    end
+  end
+
+  def get_unscheduled_hearings(appeal)
+    open_schedule_hearing_tasks = appeal.tasks.open.of_type(:ScheduleHearingTask)
+    return "" if open_schedule_hearing_tasks.nil?
+
+    regional_office = appeal.closest_regional_office
+    ro_label = regional_office.nil? ? COPY::UNKNOWN_REGIONAL_OFFICE : get_ro_label(regional_office)
+
+    unscheduled_hearings = open_schedule_hearing_tasks.map do |sh_task|
+      unscheduled_hearing_label = "Unscheduled Hearing (SCH Task ID: #{sh_task.id}, RO queue: #{ro_label})"
+      unscheduled_note = show_pii ? sh_task.parent&.instructions : "Show PII to see notes"
+
+      { unscheduled_hearing_label => ["Notes: #{print_nil(unscheduled_note)}"] }
+    end
+
+    unscheduled_hearings
+  end
+
+  def shared_appeal_children(appeal)
+    children = []
+    if appeal.appellant_is_not_veteran
+      children << "Appellant is not veteran - #{appeal&.appellant_relationship}"
+    end
+
+    children += appeal.hearings.map { |hearing| structure(hearing) }
+    children += get_unscheduled_hearings(appeal)
+
+    children << {
+      "Appeal History": get_appeal_history(appeal)
+    }
+    children
+  end
+
+  def legacy_appeal_details(legacy_appeal)
+    ["VACOLS ID: #{legacy_appeal.vacols_id}"]
+  end
+
+  def legacy_appeal_children(legacy_appeal)
+    shared_appeal_children(legacy_appeal)
+  end
+
+  def legacy_appeal_context(legacy_appeal)
+    legacy_appeal.veteran
+  end
+
+  def appeal_details(appeal)
+    ["UUID: #{appeal.uuid}"]
+  end
+
+  def appeal_children(appeal)
+    shared_appeal_children(appeal)
+  end
+
+  def appeal_context(appeal)
+    appeal.veteran
+  end
+
+  def scheduled_for(hearing)
+    ro_time = hearing.scheduled_for.in_time_zone(hearing.regional_office_timezone)
+    if ro_time != hearing.scheduled_for
+      "Scheduled for: #{readable_date(hearing.scheduled_for)} / #{ro_time.strftime('%I:%M%p %Z')}(RO time)"
+    end
+
+    "Scheduled for: #{readable_date(hearing.scheduled_for)}"
+  end
+
+  def print_nil(obj)
+    obj.presence || "none"
+  end
+
+  def get_ro_label(regional_office, request_type = false)
+    regional_office_city_info ||= RegionalOffice::CITIES[regional_office]
+    city_state_string = if regional_office_city_info.nil?
+                          ""
+                        else
+                          "#{regional_office_city_info[:city]} #{regional_office_city_info[:state]}"
+                        end
+
+    formatted_text = regional_office.nil? ? "no ro, " : "#{regional_office} - #{city_state_string}"
+    formatted_text += request_type ? ", #{Hearing::HEARING_TYPES[request_type&.to_sym]}" : ""
```

Comment:
noticed an extra space after `no ro` while testing
<img width="442" alt="Screen Shot 2021-06-10 at 2 18 44 PM" src="https://user-images.githubusercontent.com/20735998/121576795-c6d64100-c9f6-11eb-9ee1-43b1743757b9.png">

```suggestion
    formatted_text = regional_office.nil? ? "no ro," : "#{regional_office} - #{city_state_string},"
    formatted_text += request_type ? " #{Hearing::HEARING_TYPES[request_type&.to_sym]}" : " No request type"
```